pex/pex.py

```diff
@@ -348,7 +346,9 @@ def execute(self):
       self.patch_sys(pex_inherit_path)
       working_set = self._activate()
       self.patch_pkg_resources(working_set)
-      self._wrap_coverage(self._wrap_profiling, self._execute)
+      exit_code = self._wrap_coverage(self._wrap_profiling, self._execute)
```

Comment:
This was never correct. Who knows is the entry point function even returns an int? The real problem was ever using the pytest:main entrypoint. We should have been using `-c pytest` or else `-m pytest`. Pex needs to provide some guidance in its --help and docs, but using an entrypoint function (foo.bar:baz) implies you know the function does its own sys.exits to indicate failure or else raises. If it just returns some value, whether an int or a string or a what have you, PEX really can't just presume and pas that to sys.exit.

Reply:
The right path may be to keep this behavior but actually document it. Pex could say "If you pass an entrypoint _function_ and it returns anything, we'll assume that's an int that represents the exit code of the process that should be used."

It looks like sys.exit behavior is a bit whacky and gracefully supports the present use :/:
```
$ python -c 'import sys; sys.exit("Hello")'; echo $?
Hello
1
$ python -c 'import sys; sys.exit([])'; echo $?
[]
1
$ python -c 'import sys; sys.exit(["a"])'; echo $?
['a']
1
$ python -c 'import sys; sys.exit(True)'; echo $?
1
$ python -c 'import sys; sys.exit(False)'; echo $?
0
```